moveit_ros/planning/moveit_cpp/src/planning_component.cpp

```diff
@@ -308,6 +308,7 @@ bool PlanningComponent::setStartState(const std::string& start_state_name)
     return false;
   }
   moveit::core::RobotState start_state(moveit_cpp_->getRobotModel());
+  start_state.setToDefaultValues();  // required to ensure all joints are initialized
```

Comment:
I don't think this's needed, what if the current state of the gripper is open, wouldn't this cause the planner to plan with that state rather than the actual one?

Reply:
I was encountering a segfault without this setting early on when writing code for the library. There's a related discussion on Discord for this. 

@captain-yoshi encountered a similar issue [here](https://github.com/ros-planning/moveit/pull/3008).

Reply:
@JafarAbdi do you think it is reasonable to resolve this item with better documentation?

Reply:
Having to use this in multiple places makes me think that we have a deeper bug, I'm more than happy to debug it if you have a small reproducible example.

I think we should merge this PR and handle that issue in a separate one!

Reply:
Usually you would want to copy and modify some sort of 'current' `RobotState` instead of creating a new one.
(I haven't looked at this code, just read the discussion)

Reply:
> I think we should merge this PR and handle that issue in a separate one!

Happy to open up a separate issue for this and address it there. 

> Usually you would want to copy and modify some sort of 'current' RobotState instead of creating a new one.
(I haven't looked at this code, just read the discussion)

This makes sense, currently we call the getRobotState on the moveit_cpp node to retrieve the robot state before setting the state of certain joints. Joints not within the planning_group we are setting joints for sometimes can have uninitialised values which results in the errors we saw in the past :(. 

@v4hn suggested the following in the past:

> but it's really bad that we do not have proper error messages for these cases :/
So if these initializations/updates fix your problem consider writing a small patch that yields a proper error message (or just works) when someone tries to do it again

I think having a proper error messages is important, it is likely bad to assume that the user only cares about the state of the planning_group when setting the start state. We should probably check if joints are uninitialised and print some sort of message to notify the user that we set them to default values as they were uninitialised.

Reply:
I opened the following [issue](https://github.com/ros-planning/moveit2/issues/1972). 

It might also be worth making @henningkayser aware of this as it relates to ``moveit_cpp``.

Reply:
I think this solution is perfectly fine. The values are not initialized to default values because of performance reasons. In Python this doesn't exist, afaik, so always using default values is the way to go.